internal/cloud/backend.go

```diff
@@ -569,6 +569,23 @@ func (b *Cloud) StateMgr(name string) (statemgr.Full, error) {
 		}
 	}
 
+	if b.WorkspaceMapping.Strategy() == WorkspaceTagsStrategy {
+		var tags []*tfe.Tag
+		for _, tag := range b.WorkspaceMapping.Tags {
+			t := tfe.Tag{Name: tag}
+			tags = append(tags, &t)
+		}
+		options := tfe.WorkspaceAddTagsOptions{
+			Tags: tags,
+		}
```

Comment:
Considering this is called every time a state needs to be fetched, I think this should probably check the workspace for the tags and only make a call if tags are missing.

Reply:
Yup! Mentioned this above 
>  This should only occur when the Tag is not present. If the Workspace already has the Tags, no need to do an extra update